src/util/ecdsa.rs

```diff
@@ -100,7 +102,14 @@ impl PublicKey {
         };
 
         reader.read_exact(&mut bytes[1..])?;
-        Self::from_slice(bytes).map_err(|e| io::Error::new(io::ErrorKind::InvalidData, e))
+        Self::from_slice(bytes).map_err(|e|{
+            // Need a static string for core2
+            let reason = match e {
+                Error::Base58(_) => "base58 error",
```

Comment:
This was done differently depending on `std` or `no-std` in another place. Is there any particular reasoning behind trying to covey the inner error message when possible there and not here?

Reply:
I missed this one when reading the original PR, thank you.